### PR TITLE
Updates GoDocs for CLI Options and addRoute Function

### PR DESCRIPTION
**Pull request by FilledEther20**
## Pull Request

**Description:**

-   This PR introduces updates to the CLI options in the `gofr` package by enhancing the documentation for the `Options` function and the `addRoute` method. 
-   #993 is the issue being tackled through this pull request.
-   It adds clearer comments for improved understanding and accessibility, aligning with the overall documentation standards.

**Breaking Changes (if applicable):**

-   N/A

**Additional Information:**

-   N/A

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank You**

### CODE REVIEW

**Comment by Umang01-hash** [comment]
@FilledEther20 your PR is failing the code quality step in pipeline due to this error:
Screenshot 2024-10-04 at 11 51 32 AM

Please address it

File: pkg/gofr/cmd.go [hunk 1]
```diff
@@ -106,6 +108,7 @@ func (cmd *cmd) handler(path string) *route {
 	return nil
 }
 
+// Option functions to configure the service
```
Comment (aryanmehrotra) [inline pkg/gofr/cmd.go:111]: can be repharased to 

`AddDescription is an Option that adds the description text for a specified subcommand.`

File: pkg/gofr/cmd.go [hunk 2]
```diff
@@ -121,6 +124,8 @@ func AddHelp(helperString string) Options {
 	}
 }
 
+// addRoute adds a new route to cmd's list of routes.
+// Optional configuration for the route is passed using functional options like AddHelp and AddDescription.
```
Comment (aryanmehrotra) [inline pkg/gofr/cmd.go:128]: `addRoute adds a new route to cmd's list of routes.` having this is sufficient

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]

File: pkg/gofr/cmd.go (hunk 2 above)
Comment (FilledEther20) [reply pkg/gofr/cmd.go]: Ok sir have addressed this in the new commit.

**Review by aryanmehrotra — APPROVED** [review]

**Pull request merged**